source/server/options_impl_platform_linux.cc

```diff
@@ -0,0 +1,49 @@
+#include <sys/types.h>
+#include <unistd.h>
+
+#include <bitset>
+#include <fstream>
+#include <iostream>
+#include <string>
+#include <thread>
+
+#include "server/options_impl_platform.h"
+
+namespace Envoy {
+
+uint32_t OptionsImplPlatform::getCpuCount() {
```

Comment:
Can we get some test coverage of this code? Potentially a test that only runs in Linux? If you want to test it using mocks you can also consider using OsSysCalls (that might be good also to test the edge cases).

Reply:
I added some tests. I chose to refactor the `getCpuCount()` function into two to make the testing easier.